### PR TITLE
feat: Support redirects for course pages

### LINKED ISSUE #795: Support new alias for course pages
See https://github.com/serlo/db-migrations/issues/338

New url alias: /{subject}/{course-id}/{course-page-id}/{slug-from-course-page-title}

ToDos: Add redirect for course pages already in the new link format when title has changed

### PR DESCRIPTION
**Pull request by kulla**
Fixes #795 

Blocker: This PR needs https://github.com/serlo/api.serlo.org/pull/1579

### CODE REVIEW

File: __tests__/__utils__/services/database.ts [hunk 1]
```diff
@@ -12,7 +12,9 @@ export function givenUuid(uuid: Uuid) {
 export function getUuid(instance: Instance, path: string) {
   const regexes = [
     new RegExp('^/(?<id>\\d+)$'),
-    new RegExp('(?<subject>[^/]+/)?(?<id>\\d+)/(?<title>[^/]*)$'),
+    new RegExp(
+      '(?<subject>[^/]+/)?(?<id>\\d+)(?<coursePageId>/[a-z0-9-]+)?/(?<title>[^/]*)$',
```
Comment (elbotho) [inline __tests__/__utils__/services/database.ts:16]:
```suggestion
      '(?<subject>[^/]+/)?(?<id>\\d+)(?<coursePageId>/[0-9a-f]+)?/(?<title>[^/]*)$',
```
(see slack)

File: __tests__/current-alias-redirects.ts [hunk 2]
```diff
@@ -68,65 +68,52 @@ test('no redirect when requested entity has no alias', async () => {
...
+      pathname: '/math/42/page2/xyz',
     })
 
-    const response = await env.fetch({ subdomain: 'de', pathname: '/19479' })
-
     const target = env.createUrl({
-      subdomain: 'de',
-      pathname: '/mathe/20368/%C3%BCberblick',
+      subdomain: 'en',
+      pathname: '/math/42/page2/bar',
```
Comment (elbotho) [inline __tests__/current-alias-redirects.ts]:
```suggestion
      pathname: '/math/42/1a2b3c4d/bar',
```
(optional) would be more realistic

File: src/current-alias-redirects.ts [hunk 3]
```diff
@@ -116,15 +129,49 @@ async function getPathInfo(
   if (!ApiResult.is(apiResponseBody)) return null
   const uuid = apiResponseBody.data.uuid
 
+  const coursePageMatch = path.match(
+    /^\/(?<instance>[a-z]{2}\/)?(?<subject>[a-z]+\/)?(?<id>\d+)\/(?<coursePageId>[a-z0-9-]+)\/(?<title>[^/]*)$/,
```
Comment (elbotho) [inline src/current-alias-redirects.ts]:
```suggestion
    /^\/(?<instance>[a-z]{2}\/)?(?<subject>[a-z]+\/)?(?<id>\d+)\/(?<coursePageId>[0-9a-f]+)\/(?<title>[^/]*)$/,
```

**Review by elbotho — CHANGES_REQUESTED** [review]
short regex feedback, no proper review yet

**Comment by kulla** [comment]
> short regex feedback, no proper review yet

@elbotho Fixed in https://github.com/serlo/cloudflare-worker/pull/796/commits/1bd81acb4ef34002294fc2747cf9f3cf54836f9b

File: src/current-alias-redirects.ts [hunk 4]
```diff
@@ -148,3 +195,25 @@ async function getPathInfo(
 function gql(strings: TemplateStringsArray): string {
   return strings[0]
 }
+
+// Copied from https://github.com/serlo/api.serlo.org/blob/ce94045b513e59da1ddd191b498fe01f6ff6aa0a/packages/server/src/schema/uuid/abstract-uuid/resolvers.ts#L685-L703
```
Comment (AndreasHuber) [inline src/current-alias-redirects.ts:199]: Why? Can't you put it in a separate file to be used by both?

**Comment by elbotho** [comment]
@kulla what would happen if I request an old `coursePage` path like `/51553` or `/mathe/51553/linearfaktordarstellung-1-3`? Do they get redirected to the new format now?

**Comment by kulla** [comment]
> @kulla what would happen if I request an old `coursePage` path like `/51553` or `/mathe/51553/linearfaktordarstellung-1-3`? Do they get redirected to the new format now?

Should happen after the API merge

**Review by elbotho — APPROVED** [review]
I will try this in staging and address improvements / fixes later

**Pull request merged**